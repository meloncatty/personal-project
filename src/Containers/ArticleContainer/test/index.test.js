import React from 'react'
import { ArticleContainer } from '../'
import { shallow } from 'enzyme'

describe("ArticleContainer", () => {
  it("should match snapshot", () => {
    const mockArticle = {
      authors: 'author name',
      title: 'article title',
      datePublished: '01-01-2018',
      fullText: 'full article text',
      topics: 'article topics',
      fullTextIdentifier: 'link-to-fulltext.com'
    }
    const wrapper = shallow(<ArticleContainer fullArticleSuccess={mockArticle}/>)
    expect(wrapper).toMatchSnapshot()
  });

  it("should have default state", () => {
    const mockArticle = {
      authors: 'author name',
      title: 'article title',
      datePublished: '01-01-2018',
      fullText: 'full article text',
      topics: 'article topics',
      fullTextIdentifier: 'link-to-fulltext.com'
    }
    const wrapper = shallow(<ArticleContainer fullArticleSuccess={mockArticle}/>)
    const expected = {
      toggleText: false
    }

    expect(wrapper.state()).toEqual(expected)
  });

  it("should toggle text", () => {
    const mockArticle = {
      authors: 'author name',
      title: 'article title',
      datePublished: '01-01-2018',
      fullText: 'full article text',
      topics: 'article topics',
      fullTextIdentifier: 'link-to-fulltext.com'
    }
    const wrapper = shallow(<ArticleContainer fullArticleSuccess={mockArticle}/>)
    const expected = {
      toggleText: true
    }
    wrapper.find('.lbl-toggle').simulate('click')

    expect(wrapper.state()).toEqual(expected)
  });
  it("should call loading station while loading", () => {
    const mockArticle = {
      authors: 'author name',
      title: 'article title',
      datePublished: '01-01-2018',
      fullText: 'full article text',
      topics: 'article topics',
      fullTextIdentifier: 'link-to-fulltext.com'
    }
    const wrapper = shallow(<ArticleContainer fullArticleLoading={true} fullArticleSuccess={mockArticle}/>)

    expect(wrapper.instance().loadingStation).toHaveBeenCalled
  });

  it("should call loading station while loading", () => {
    const mockArticle = {
      authors: 'author name',
      title: 'article title',
      datePublished: '01-01-2018',
      fullText: 'full article text',
      topics: 'article topics',
      fullTextIdentifier: 'link-to-fulltext.com'
    }
    const wrapper = shallow(<ArticleContainer fullArticleLoading={false} fullArticleSuccess={mockArticle}/>)

    expect(wrapper.instance().cleanFullArticle).toHaveBeenCalled
  });
});
