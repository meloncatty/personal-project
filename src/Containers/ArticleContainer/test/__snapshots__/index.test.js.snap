// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ArticleContainer should match snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ArticleContainer
    fullArticleSuccess={
        Object {
            "authors": "author name",
            "datePublished": "01-01-2018",
            "fullText": "full article text",
            "fullTextIdentifier": "link-to-fulltext.com",
            "title": "article title",
            "topics": "article topics",
          }
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <article
          className="article-contents"
>
          <h2>
                    article title
          </h2>
          <h3>
                    Authors: 
                    author name
          </h3>
          <h3>
                    Date published: 
                    01-01-2018
          </h3>
          <label
                    className="lbl-toggle"
                    onClick={[Function]}
          >
                    Click to expand
          </label>
          <div
                    className="toggle-text closed"
          >
                    <p>
                              full article text
                    </p>
          </div>
          <h4>
                    Topics: 
                    article topics
          </h4>
</article>,
        undefined,
      ],
      "className": "article-container",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <h2>
              article title
</h2>,
            <h3>
              Authors: 
              author name
</h3>,
            <h3>
              Date published: 
              01-01-2018
</h3>,
            <label
              className="lbl-toggle"
              onClick={[Function]}
>
              Click to expand
</label>,
            <div
              className="toggle-text closed"
>
              <p>
                            full article text
              </p>
</div>,
            <h4>
              Topics: 
              article topics
</h4>,
            undefined,
          ],
          "className": "article-contents",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "article title",
            },
            "ref": null,
            "rendered": "article title",
            "type": "h2",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                "Authors: ",
                "author name",
              ],
            },
            "ref": null,
            "rendered": Array [
              "Authors: ",
              "author name",
            ],
            "type": "h3",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                "Date published: ",
                "01-01-2018",
              ],
            },
            "ref": null,
            "rendered": Array [
              "Date published: ",
              "01-01-2018",
            ],
            "type": "h3",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Click to expand",
              "className": "lbl-toggle",
              "onClick": [Function],
            },
            "ref": null,
            "rendered": "Click to expand",
            "type": "label",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <p>
                full article text
</p>,
              "className": "toggle-text closed",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "full article text",
              },
              "ref": null,
              "rendered": "full article text",
              "type": "p",
            },
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                "Topics: ",
                "article topics",
              ],
            },
            "ref": null,
            "rendered": Array [
              "Topics: ",
              "article topics",
            ],
            "type": "h4",
          },
          undefined,
        ],
        "type": "article",
      },
      undefined,
    ],
    "type": "section",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <article
            className="article-contents"
>
            <h2>
                        article title
            </h2>
            <h3>
                        Authors: 
                        author name
            </h3>
            <h3>
                        Date published: 
                        01-01-2018
            </h3>
            <label
                        className="lbl-toggle"
                        onClick={[Function]}
            >
                        Click to expand
            </label>
            <div
                        className="toggle-text closed"
            >
                        <p>
                                    full article text
                        </p>
            </div>
            <h4>
                        Topics: 
                        article topics
            </h4>
</article>,
          undefined,
        ],
        "className": "article-container",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <h2>
                article title
</h2>,
              <h3>
                Authors: 
                author name
</h3>,
              <h3>
                Date published: 
                01-01-2018
</h3>,
              <label
                className="lbl-toggle"
                onClick={[Function]}
>
                Click to expand
</label>,
              <div
                className="toggle-text closed"
>
                <p>
                                full article text
                </p>
</div>,
              <h4>
                Topics: 
                article topics
</h4>,
              undefined,
            ],
            "className": "article-contents",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "article title",
              },
              "ref": null,
              "rendered": "article title",
              "type": "h2",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  "Authors: ",
                  "author name",
                ],
              },
              "ref": null,
              "rendered": Array [
                "Authors: ",
                "author name",
              ],
              "type": "h3",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  "Date published: ",
                  "01-01-2018",
                ],
              },
              "ref": null,
              "rendered": Array [
                "Date published: ",
                "01-01-2018",
              ],
              "type": "h3",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Click to expand",
                "className": "lbl-toggle",
                "onClick": [Function],
              },
              "ref": null,
              "rendered": "Click to expand",
              "type": "label",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <p>
                  full article text
</p>,
                "className": "toggle-text closed",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "full article text",
                },
                "ref": null,
                "rendered": "full article text",
                "type": "p",
              },
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  "Topics: ",
                  "article topics",
                ],
              },
              "ref": null,
              "rendered": Array [
                "Topics: ",
                "article topics",
              ],
              "type": "h4",
            },
            undefined,
          ],
          "type": "article",
        },
        undefined,
      ],
      "type": "section",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
